### DISCUSSION TITLE
博士新版封面 & 国图封面 评审人下划线太短，请教如何设置

### DISCUSSION THREAD

**Discussion by dreistein543** [nju-lug/NJUThesis#221] [Q&A]
如题，由于博士生普通封面和国图封面均需要填写评审人单位，但是预留的下划线长度不够，请问该如何设置下划线长度？

普通封面信息如下
<img width="880" alt="image" src="https://github.com/nju-lug/NJUThesis/assets/30745407/d202642c-6e21-413d-8fa2-8701fa5a56f6">

国图封面信息如下
<img width="1172" alt="image" src="https://github.com/nju-lug/NJUThesis/assets/30745407/bd3b8d8b-f8d3-4ec4-a2d6-4231a26001c6">

**Comment by atxy-blip** [comment]
前者可以靠 `\njusetlength{crulewdi}{8cm}` 改，后者还没做用户接口，在 `njuthesis-graduate.def` 改一下 `\__nju_cover_bottom_nl:` 里面设定的 9em 长度吧。

**Reply by dreistein543** [reply]
不知是否做好了 ` 国图封面` 的用户接口？25年年初毕业论文入国图前最后一改好像还没加上。

P.S. 当时本想尝试最新的模板，结果发现毕业后似乎改动很多，又乖乖用回毕业前的模板了（感谢 tex.nju.edu.cn 的版本管理功能）
P.P.S 昨天惊闻GitHub被403，今天尝试一下后想起来还有当年留下的小尾巴🤣

**Reply by atxy-blip** [reply]
我还没做…… 最近学业上生活上事情比较多，比较大的改动也没有动力去开展。

**Comment by acddca** [comment]
教务老师给的模板 南大封面校徽是紫色的 请问要怎么设置

**Reply by dreistein543** [reply]
我记得模板包里是有彩色校徽的，替换一下文件就好啦

**Reply by atxy-blip** [reply]
见 `njuthesis-setup.def`，这些设置选项都在手册 `njuthesis.pdf` 中有详细的描述
```tex
% image 类用于载入外置的图片
\njusetup[image]{
    % path = {{./figure/}{./image/}},
    % 图片搜索路径
    %
    nju-emblem = {nju-emblem},
    nju-name = {nju-name},
    % 校徽和校名图片路径
    % 建议使用 PDF 格式的矢量图
    % 使用外置图片有助于减少编译时间
    % 空置时会自动使用 njuvisual 宏包绘制
    %
    % nju-emblem = {nju-emblem-purple},
    % nju-name = {nju-name-purple},
    % 替换为紫色版本
    % 这个选项只能填写一次
    % 切换时要注释掉上方的黑色版本
}
```

另：你的两个问题属于 off-topic，理应另开新帖。

**Comment by acddca** [comment]
国图封面的日期默认是，2022年5月23日 请问这个需要修改吗？如果是，该如何修改

**Reply by atxy-blip** [reply]
见 `njuthesis-setup.def`
```tex
...
    % 以下用于国家图书馆封面
    confer-date = {2022-05-22},
    % 学位授予日期
    %
    bottom-date = {2022-05-23},
    % 封面底部日期
...
```
